examples/wallet_examples/internalize_wallet_payment/internalize_wallet_payment.go

```diff
@@ -62,31 +70,77 @@ func main() {
 		panic(fmt.Errorf("failed to get sender identity key: %w", err))
 	}
 
+	keyID := brc29.KeyID{
+		DerivationPrefix: Prefix,
+		DerivationSuffix: Suffix,
+	}
+
+	aliceKeyDeriver := sdk.NewKeyDeriver(alice.PrivateKey)
+	derivedPrivateKey, err := aliceKeyDeriver.DerivePrivateKey(brc29.Protocol, keyID.String(), sdk.Counterparty{
+		Type:         sdk.CounterpartyTypeOther,
+		Counterparty: senderIdentityKey,
+	})
+	if err != nil {
+		panic(fmt.Errorf("failed to derive private key for Alice: %w", err))
+	}
+
+	derivedAddress, err := script.NewAddressFromPublicKey(derivedPrivateKey.PubKey(), isBsvNetworkMain)
+	if err != nil {
+		panic(fmt.Errorf("failed to create address from derived private key: %w", err))
+	}
+
+	addressObj, err := script.NewAddressFromString(derivedAddress.AddressString)
+	if err != nil {
+		panic(fmt.Errorf("failed to parse address %q: %w", derivedAddress, err))
+	}
```

Comment:
why you're not using brc29.Address function to get the address?
Is this code doing deriving some other address then the brc29 address?